public/app/features/explore/state/reducers.ts

```diff
@@ -512,7 +512,7 @@ export const processQueryResponse = (
     loading: loadingState === LoadingState.Loading || loadingState === LoadingState.Streaming,
     update: makeInitialUpdateState(),
     showLogs: !!logsResult,
-    showMetrics: !!graphResult,
+    showMetrics: Boolean(graphResult?.length),
```

Comment:
Not sure but I think this may prevent showing a `No data` message? I mean if we get graphResult with zero length we should probably show an empty graph or something. Similar for logs if we get logsResult with zero length maybe the logs panel should show some `No data` state. At the same time not sure how that is implemented now and maybe we have actually lost that info when unifying the interface.

Reply:
Yes you are right. Nice catch.